tests/python_client/testcases/test_query.py

```diff
@@ -654,6 +640,268 @@ def test_query_expr_json_contains_combined_with_normal(self, enable_dynamic_fiel
+
+        collection_w.insert(array)
+
+        # 3. query
+        collection_w.load()
+        expr_prefix = ["json_contains_all", "JSON_CONTAINS_ALL"]
+        for expr in expr_prefix:
+            # test for int
+            _id = random.randint(0, ct.default_nb)
+            ids = [i for i in range(_id, _id + limit)]
+            expression = f"{expr}(listInt, {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == 1
+
+            # test for string
+            ids = [str(_id), str(_id + 1), str(_id + 2)]
+            expression = f"{expr}(listStr, {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == limit - len(ids) + 1
+
+            # test for float
+            ids = [_id * 1.0]
+            expression = f"{expr}(listFlt, {ids})"
+            res = collection_w.query(expression, output_fields=["count(*)"])[0]
+            assert res[0]["count(*)"] == limit
+
+            # test for bool
+            ids = [True]
+            expression = f"{expr}(listBool, {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == ct.default_nb // 2
+
+            # test for list
+            ids = [[_id, str(_id + 1)]]
+            expression = f"{expr}(listList, {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == 1
+
+            # test for mixed data
+            ids = [_id * 1.1, bool(_id % 2)]
+            expression = f"{expr}(listMix, {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == 1
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_expr_all_datatype_json_contains_any(self, enable_dynamic_field):
+        """
+        target: test query with expression using json_contains
+        method: query with expression using json_contains
+        expected: succeed
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, enable_dynamic_field=enable_dynamic_field)[0]
+
+        # 2. insert data
+        array = cf.gen_default_rows_data()
+        limit = 10
+        for i in range(ct.default_nb):
+            content = {
+                "listInt": [m for m in range(i, i + limit)],  # test for int
+                "listStr": [str(m) for m in range(i, i + limit)],  # test for string
+                "listFlt": [m * 1.0 for m in range(i, i + limit)],  # test for float
+                "listBool": [bool(i % 2)],  # test for bool
+                "listList": [[i, str(i + 1)], [i * 1.0, i + 1]],  # test for list
+                "listMix": [i, i * 1.1, str(i), bool(i % 2), [i, str(i)]]  # test for mixed data
+            }
+            array[i][ct.default_json_field_name] = content
+
+        collection_w.insert(array)
+
+        # 3. query
+        collection_w.load()
+        expr_prefix = ["json_contains_any", "JSON_CONTAINS_ANY"]
```

Comment:
I think it is better to use fixture to replace the for loop.

Reply:
It is a good idea. I have modified all the case.